src/globus_cli/commands/gcs/endpoint/set_subscription_id.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.constants import EXPLICIT_NULL, ExplicitNullType
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.termio import TextMode, display
+
+
+class GCSSubscriptionIdType(click.ParamType):
+    def get_type_annotation(self, _: click.Parameter) -> type:
+        return t.cast(type, uuid.UUID | t.Literal["DEFAULT"] | ExplicitNullType)
+
+    def convert(
+        self, value: str, param: click.Parameter | None, ctx: click.Context | None
+    ) -> t.Any:
+        if value is None or (ctx and ctx.resilient_parsing):
+            return None
+        if value.lower() == "null":
+            return EXPLICIT_NULL
+        elif value.lower() == "default":
+            return "DEFAULT"
+        try:
+            uuid.UUID(value)
+            return value
+        except ValueError:
+            msg = (
+                f"{value} is invalid. Expected either a UUID or the special "
+                'values "DEFAULT" or "null"'
+            )
+            self.fail(msg, param, ctx)
+
+
+@command("set-subscription-id", short_help="Update an endpoint's subscription")
+@endpoint_id_arg
+@click.argument("SUBSCRIPTION_ID", type=GCSSubscriptionIdType())
+@LoginManager.requires_login("transfer")
+def set_subscription_id_command(
+    login_manager: LoginManager,
+    *,
+    endpoint_id: uuid.UUID,
+    subscription_id: uuid.UUID | t.Literal["DEFAULT"] | ExplicitNullType,
+) -> None:
+    """
+    Update an endpoint's subscription ID.
+
+    SUBSCRIPTION_ID must be one of: (1) A valid subscription ID (UUID), (2) the value
+    "DEFAULT" (requires that you manage exactly one subscription & assigns the endpoint
+    to that subscription), or (3) the value "null" (clears the endpoint's subscription)
+
+    Setting a subscription requires that you are a subscription manager for the
+    subscription group being assigned.
```

Comment:
The phrase "subscription group" catches my eye. Just want to confirm that this terminology is correct.

Reply:
Posted this Q to auth for a more authoritative answer:

https://globus.slack.com/archives/C03N0A8J7/p1703102225633449

Reply:
Updated to ```Setting a subscription requires that you are a subscription manager for the subscription being assigned.``` based on Josh's response.